src/main/java/duke/component/Parser.java

```diff
@@ -0,0 +1,89 @@
+package duke.component;
+
+import duke.command.*;
+
+import duke.exception.EmptyDescriptionException;
+import duke.exception.UnknownCommandException;
+import duke.exception.WrongFormatException;
+import duke.task.ToDo;
+import duke.task.Deadline;
+import duke.task.Event;
+
+public class Parser {
+    /**
+     * Parses command and generate task accordingly.
+     * @param command
+     * @return
+     * @throws UnknownCommandException
+     * @throws EmptyDescriptionException
+     * @throws WrongFormatException
+     */
+    public static Command parse(String command) throws UnknownCommandException, EmptyDescriptionException, WrongFormatException {
+        String[] parameters = command.split(" ");
+        switch (parameters[0]) {
+        case "bye": {
+            return new ExitCommand();
+        }
+        case "list": {
+            return new ListCommand();
+        }
+        case "done": {
+            if (command.length() <= 5) {
+                throw new WrongFormatException();
+            }
+            return new DoneCommand(Integer.parseInt(parameters[1]) - 1);
+        }
+        case "delete": {
+            if (command.length() <= 7) {
+                throw new WrongFormatException();
+            }
+            return new DeleteCommand(Integer.parseInt(parameters[1]) - 1);
+        }
+        case "todo": {
+            if (command.trim().length() <= 4) {
+                throw new EmptyDescriptionException("todo");
+            }
+            ToDo td = new ToDo(command.substring(5, command.length()));
+            return new AddCommand(td);
+        }
+        case "deadline": {
```

Comment:
Case blocks for "deadline" and "event" look quite long. Maybe abstracting out the code segments for handling input description and datetime could help to reduce length of case block and also place the code segments under method names that can be easily found if further editing is needed?

Reply:
Thanks for the input! I have refactored accordingly.